### DISCUSSION TITLE
SOLVED: Nightscout on Google Cloud Issue: Remote overrides error

### DISCUSSION THREAD

**Discussion by mtndev** [NightscoutFoundation/xDrip#2678] [General]
Hi- I've been using the Google Cloud Nightscout with no problems for months now. I haven't changed anything. Then this morning when I needed to turn on a remote override while my kid is at school, it wouldn't work. There's an error that says:
"Error: APNs delivery failed: InvalidProviderToken"

it's in a popup that happens when I click the Submit Form button. I've tried in different browsers, also desktop/mobile. I tried other commands besides remote overrides and those work as expected.

Here's a screencapture: https://prnt.sc/BMV1mVP6r1sw

I searched and there's a reference to this error in the loopdocs: https://loopkit.github.io/loopdocs/nightscout/remote-overrides/#:~:text=Error%3A%20APNs%20delivery%20failed%3A%20InvalidProviderToken,the%20config%20variables%20in%20Heroku.

It says "This is caused because your LOOP_APNS_KEY_ID and LOOP_DEVELOPER_TEAM_ID are swapped in Heroku"

However, I haven't changed anything in the varialbes since I built the server months ago. Since the config variables worked fine it seems like perhaps there's a different cause. Any ideas?

**Comment by Navid200** [comment]
@tzachi-dar Would you please have a look at this?

**Comment by Navid200** [comment]
@mtndev This is just a test.
This may not make any difference.

Edit variables and change the following
AUTH_DEFAULT_ROLES="denied"
to
AUTH_DEFAULT_ROLES="readable"

Reboot the server.

**Reply by mtndev** [reply]
I did that but it didn't help. Any other ideas?

Here's a screencap of the variables now:
https://prnt.sc/E6K2rTxK2iBl

**Comment by tzachi-dar** [comment]
Is this all the keys that you have? you are missing the key for the LOOP_APNS_KEY_ID.
In any case, can you please try to create this keys again and see if that will solve the issue?

**Reply by mtndev** [reply]
Hi- recreating the keys did work. I didn't get them pasted in correctly the first time so was really confused for awhile. But i saw the mistake now and got it fixed. New keys worked. Thank you!
